src/melee.cpp

```diff
@@ -1940,3 +1945,124 @@ double player::unarmed_value() const
     // TODO: Martial arts
     return melee_value( ret_null );
 }
+
+void player::disarm( npc &target )
+{
+    if( !target.is_armed() ) {
+        return;
+    }
+
+    item &it = target.weapon;
+
+    static const matec_id no_technique_id( "" );
+
+    ///\EFFECT_STR increases chance to disarm, primary stat
+    ///\EFFECT_DEX increases chance to disarm, secondary stat
+    int my_roll = dice( 3, 2 * get_str() + get_dex() );
+
+    ///\EFFECT_MELEE increases chance to disarm
+    my_roll += dice( 3, get_skill_level( skill_melee ) );
+
+    int their_roll = dice( 3, 2 * target.get_str() + target.get_dex() );
+    their_roll += dice( 3, target.get_per() );
+    their_roll += dice( 3, target.get_skill_level( skill_melee ) );
+
+    // make the target angry as they confess our unfriendly move
+    target.make_angry();
+
+    // roll your melee and target's dodge skills to check if grab/smash attack succeeds
+    int hitspread = target.deal_melee_attack( this, hit_roll() );
+    if( hitspread < 0 ) {
+        // this will not do damage, but trigger all miss effects and on_dodge on target
+        melee_attack( target, false, no_technique_id, hitspread );
+        return;
+    }
+
+    // hitspread >= 0, which means we are going to disarm by grabbing target by their weapon
+    if( !is_armed() ) {
+        ///\EFFECT_UNARMED increases chance to disarm, bonus when nothing wielded
+        my_roll += dice( 3, get_skill_level( skill_unarmed ) );
+
+        if( my_roll >= their_roll ) {
+            add_msg( _( "You grab at %s and pull with all your force!" ), it.tname().c_str() );
+            add_msg( _( "You forcefully take %s from %s!" ), it.tname().c_str(), target.name.c_str() );
+            // wield() will deduce our moves, consider to deduce more/less moves for balance
+            item rem_it = target.i_rem( &it );
+            wield( rem_it );
```

Comment:
Unable to inline this:

src/melee.cpp:1989:20: error: non-const lvalue reference to type 'item' cannot
      bind to a temporary of type 'item'
            wield( target.i_rem( &it ) );
                   ^~~~~~~~~~~~~~~~~~~
src/player.h:860:35: note: passing argument to parameter 'target' here
        virtual bool wield( item& target );